src/main/java/graphql/util/NodeZipper.java

```diff
@@ -99,34 +101,38 @@ public T toRoot() {
         T curNode = this.curNode;
 
         Breadcrumb<T> firstBreadcrumb = breadcrumbs.get(0);
-        Map<String, List<T>> childrenForParent = nodeAdapter.getNamedChildren(firstBreadcrumb.getNode());
+        Map<String, List<T>> childrenForParent = new LinkedHashMap<>(nodeAdapter.getNamedChildren(firstBreadcrumb.getNode()));
```

Comment:
Linked hasmap again

Does it really require it ?  Its set back into the node yes as new children - but does the order of children matter

Or is this too a step too far?